plugins/inputs/xtremio/xtremio.go

```diff
@@ -0,0 +1,485 @@
+package xtremio
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type XtremIO struct {
+	Username   string   `toml:"username"`
+	Password   string   `toml:"password"`
+	URL        string   `toml:"url"`
+	Collectors []string `toml:"collectors"`
+	cookie     *http.Cookie
+	client     *http.Client
+	tls.ClientConfig
+	Log telegraf.Logger `toml:"-"`
+}
+
+type BBU struct {
+	Content struct {
+		Serial       string `json:"serial-number"`
+		Guid         string `json:"guid"`
+		PowerFeed    string `json:"power-feed"`
+		Name         string `json:"Name"`
+		ModelName    string `json:"model-name"`
+		BBUPower     int    `json:"power"`
+		BBUDailyTemp int    `json:"avg-daily-temp"`
+		BBUEnabled   string `json:"enabled-state"`
+		BBUNeedBat   string `json:"ups-need-battery-replacement"`
+		BBULowBat    string `json:"is-low-battery-no-input"`
```

Comment:
Is this really a string? From the code below it looks more like a boolean?

Reply:
It comes in as a string in the response, but I can cast it to a bool here. Probably makes more sense that way.